### ISSUE TITLE
Symbols don't resolve when /proc not available on Linux

### ISSUE THREAD

**Issue by cameronelliott** [rust-lang/backtrace-rs#636]
TL;DR: On Linux, if /proc is missing or unreadable, your symbols will not resolve. ie `BacktraceFrame.symbols()` will return an empty slice.

I spent a lot of time figuring out why symbols wouldn't resolve in a `chroot` environment on Linux,
and couldn't find any help about why it might be happening in the issues.

(I was quiet confused, because I though all symbol info was contained in the dwarf section my statically linked binary, but obviously backtrace-rs was failing to resolve symbols for some reason!)

When I finally figured out what might be going on, and mounted the /proc folder inside my chroot enviroment,
and symbol resolution started working again.

There is no request here, but I think this issue might be helpful for others who are having symbols failing to resolve due to /proc not being present in a non-traditional situation.

**Comment by cameronelliott** [comment]
No action needed or requested, but this issue is for those where /proc is missing, and symbols don't resolve.

**Comment by cameronelliott** [comment]
This line appears to where the dependency on /proc occurs:
[        File::open("/proc/self/maps").map_err(|_| "Couldn't open /proc/self/maps")?;](https://github.com/rust-lang/backtrace-rs/blob/72265bea210891ae47bbe6d4f17b493ef0606619/src/symbolize/gimli/parse_running_mmaps_unix.rs#L59)
